### PR TITLE
Incorporating Phil's changes to get SSL working again.

### PR DESCRIPTION
**Pull request by absurdfarce**
This commit represents a (slightly modified) version of commits from e3adcd0ff45ed029ffd7d68a2152e587e3b46a89
at https://github.com/philip-doctor/cassaforte.  The only change from those commits is that the project owner
here is absurdfarce... pretty sure that's necessary to keep clojars happy.

### CODE REVIEW

**Comment by absurdfarce** [comment]
@philip-doctor I think I got everything from your branch... please double-check me and make sure I didn't miss anything.

File: src/clojure/clojurewerkz/cassaforte/client.clj [hunk 1]
```diff
@@ -128,14 +128,15 @@
...
-                            { "TLS_RSA_WITH_AES_128_CBC_SHA", "TLS_RSA_WITH_AES_256_CBC_SHA" })]
+                            (into-array String ["TLS_RSA_WITH_AES_128_CBC_SHA" "TLS_RSA_WITH_AES_256_CBC_SHA"]))]
     (.load keystore keystore-stream password)
     (.init keymanager keystore password)
     (.init trustmanager keystore)
     (.init ssl-context (.getKeyManagers keymanager) (.getTrustManagers trustmanager) nil)
-    (-> (JdkSSLOptions/builder)
-        (.withSSLContext ssl-context)
-        (.withCipherSuites ssl-cipher-suites))))
+    (.. (JdkSSLOptions$Builder.)
```
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: So I get (and prefer) the .. syntax here... but why the change to instantiating a Builder class directly?  JdkSSLOptions/builder does the same thing, doesn't it?

File: src/clojure/clojurewerkz/cassaforte/client.clj [hunk 2]
```diff
@@ -116,7 +116,7 @@
       (.withAuthProvider builder (DseAuthProvider.)))
     (.build builder)))
 
-(defn- ^SSLOptions build-ssl-options
+(defn- ^JdkSSLOptions build-ssl-options
```
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: Is this necessary for type hints or something?  Seems like we'd _want_ to keep the more generic parent class here unless Clojure forces us to return the more specific type.

File: project.clj [hunk 3]
```diff
@@ -1,4 +1,4 @@
-(defproject absurdfarce/cassaforte "2.0.1-30drivers-1.0.0"
+(defproject absurdfarce/cassaforte "2.0.1-30drivers-1.0.2"
```
Comment (absurdfarce) [inline project.clj]: I preserved version numbers here since this represents identical code; a version is a version, whether it's in your repo or mine. :)

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 1 above)
Comment (philip-doctor) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: I'm fine with going back to (JdkSSLOptions/builder) it's probably a better choice in case DS java team starts adding logic to that method anyhow, so +1, the change I was debugging here is that we needed to (build) as a part of this and we were missing that step.

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 2 above)
Comment (philip-doctor) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: I was debugging a type coercion error, I thought it was related to the return type but it was actually related to the lack of (build) call (see below).  We can go back to SSLOptions, I think that's fine to use the more general interface instead.  I think I meant to clean this up when I was done debugging and just forgot.

File: src/clojure/clojurewerkz/cassaforte/client.clj [hunk 4]
```diff
@@ -26,7 +26,7 @@
             [qbits.hayt.cql :as hayt])
   (:import [com.datastax.driver.core Statement ResultSet ResultSetFuture Host Session Cluster
             Cluster$Builder SimpleStatement PreparedStatement HostDistance PoolingOptions
-            SSLOptions JdkSSLOptions ProtocolOptions$Compression ProtocolVersion]
+            SSLOptions JdkSSLOptions JdkSSLOptions$Builder ProtocolOptions$Compression ProtocolVersion]
```
Comment (philip-doctor) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: if you go back to /build then you can kill this import too.  Thanks.

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 1 above)
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: Yeah, the missing build() call was... horrible.  I have no idea how I missed that one... thanks for catching it!

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 2 above)
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: Okay, I'll probably make this change (and the change to go back to the builder() call referenced above) before pulling all this in.

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 4 above)
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: +1

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 1 above)
Comment (philip-doctor) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: no problem, I'm here to try to fix a bug, not pass judgement, don't worry about it :)

**Comment by absurdfarce** [comment]
@philip-doctor Apologies, the day got away from me a bit today.  I think I covered everything we wanted to change here... can you take one more pass and make sure you're happy with what's here?  If you are then I'll push up a new JAR containing these changes.
